Visual Studio Configure Setup and Debugging (#3116)
godotengine/godot-docs: development/cpp/configuring_an_ide.rst

- It should break at the point!

.. _doc_configuring_an_ide_vs:

Visual Studio
--------------

> Visual Studio Community is free for use and has lots of great features, such as memory view, performance view, source control and more.

Reviewer: ```suggestion
Visual Studio Community is free for non-commercial use and has lots of great features, such as memory view, performance view, source control and more.
```

(It can be free for commercial use in some cases, but let's keep it simple.)

You can get it `at microsoft <https://visualstudio.microsoft.com/downloads/>`_.

Setup
^^^^^^

> In order to start developing with this IDE please follow:

Reviewer: ```suggestion
To start developing with Visual Studio, follow these steps:
```

> - Open Visual Studio Installer and install the C++ package to start working with C++

Reviewer: ```suggestion
- Open the Visual Studio Installer and install the C++ package:
```

.. image:: img/vs_1_install_cpp_package.png

> - Use cmd or powershell to configure the project to generate ``.sln`` files and stuff and run: ``scons p=windows vsproj=yes``

Reviewer: ```suggestion
- Open a Command Prompt or PowerShell window, use ``cd`` to reach the Godot source directory and run ``scons platform=windows vsproj=yes``.
```

> - Now open the godot folder by clicking "Open a project or solution" and open ``godot.sln``

Reviewer: ```suggestion
- Now open the Godot folder by clicking **Open a project or solution** and choose ``godot.sln``.
  - You can also double-click the ``godot.sln`` file in Explorer.
```

You can now start developing with Visual Studio.

Debugging
^^^^^^^^^^

> Debugging with Visual Studio is easy. This allows the user to examin the source code of godot, stop at spesific points in the code,

Reviewer: ```suggestion
Visual Studio features a powerful debugger. This allows the user to examine Godot's source code, stop at specific points in the code,
```

make changes, and view them on the run.

.. Note:: Debugging the Godot Engine inside the editor will require an extra setup step.
             

>              Because openning godot opens firstly the Project Manager, and not the project(game) you are working on,

Reviewer: ```suggestion
             Because opening Godot opens the Project Manager at first instead of the project you're working on,
```

>              the debugger will detach as soon as you open a project, meanning the debugger will stop, even though godot is still running.

Reviewer: ```suggestion
             the debugger will detach as soon as you open a project. This means the debugger will stop, even though Godot is still running in another process.
```

>              The reason is because the debugger attaches to the Project Manager process, and when you open a project it will close and open

Reviewer: ```suggestion
             The reason is that the debugger attaches to the Project Manager process only. When you open a project, it will close and open another process for
```

             the editor.
             

> To overcome this, you need to edit the debugging command line arguments in VS. In your project, click *Project* -> *Project Properties*:

Reviewer: ```suggestion
To overcome this, you need to edit the debugging command line arguments in VS. In your project, click **Project > Project Properties**:
```

.. image:: img/vs_2_project_properties.png

> And add this to command arguments:

Reviewer: ```suggestion
Then add this to command arguments:
```

.. image:: img/vs_3_debug_command_line.png

> - The ``-e`` flag is to enter the Editor mode, skip the Project Manager.

Reviewer: ```suggestion
- The ``-e`` flag is for entering the editor directly (which skips the Project Manager).
```

> - The ``--path`` argument is what project to open.

Reviewer: ```suggestion
- The ``--path`` argument should be an *absolute* path to a project directory (not a `project.godot` file).
```

> To learn more about command line arguments, refer to  :ref:`command line tutorial <doc_command_line_tutorial>`

Reviewer: ```suggestion
To learn more about command line arguments, refer to the :ref:`command line tutorial <doc_command_line_tutorial>`.
```

> To check that everything is working, put breakpoint in main.cpp and press F5 to start debugging.

Reviewer: ```suggestion
To check that everything is working, put a breakpoint in ``main.cpp`` and press F5 to start debugging.
```
